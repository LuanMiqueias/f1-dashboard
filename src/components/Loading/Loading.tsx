import React from "react";
import styles from "./style.module.css";

const Loading = () => {
  return (
    <div className={styles.container} data-testid="loading">
      <div className={styles.loading}></div>
    </div>
  );
};

export default Loading;
