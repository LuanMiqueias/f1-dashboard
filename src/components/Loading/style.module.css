.loading{
  width: 45px;
  height: 45px;
  border-radius: 50%;
  border: solid var(--gray-light);
  border-top: solid var(--primary-color);
  border-width: 1px;
  animation: loadingAnima .5s linear infinite;
}
@keyframes loadingAnima{
  to{
    transform: rotate(360deg);
  }
}
.container{
  width: 100%;
  height: 150px;
  display: flex;
  align-items: center;
  justify-content: center;
}