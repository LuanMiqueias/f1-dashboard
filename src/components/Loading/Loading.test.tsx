import { render } from "@testing-library/react";
import Loading from "./Loading";

it("Should math snapshot", () => {
  const { container } = render(<Loading />);

  expect(container.firstChild).toMatchSnapshot();
});
