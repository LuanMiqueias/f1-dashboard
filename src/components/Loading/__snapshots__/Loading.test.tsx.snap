// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should math snapshot 1`] = `
<div
  class="container"
>
  <div
    class="loading"
  />
</div>
`;
